packages/google_maps_flutter/google_maps_flutter/example/lib/map_map_id.dart

```diff
@@ -0,0 +1,145 @@
+  const MapIdBody({super.key});
+
+  @override
+  State<StatefulWidget> createState() => MapIdBodyState();
+}
+
+const LatLng _kMapCenter = LatLng(52.4478, -3.5402);
+
+class MapIdBodyState extends State<MapIdBody> {
+  GoogleMapController? controller;
+
+  Key _key = const Key('mapId#');
+  String? _mapId;
+  final TextEditingController _mapIdController = TextEditingController();
+  AndroidMapRenderer? _initializedRenderer;
+
+  @override
+  void initState() {
+    initializeMapRenderer()
+        .then<void>((AndroidMapRenderer? initializedRenderer) => setState(() {
+              _initializedRenderer = initializedRenderer;
+            }));
+    super.initState();
+  }
+
+  String _getInitializedsRendererType() {
+    switch (_initializedRenderer) {
+      case AndroidMapRenderer.latest:
+        return 'latest';
+      case AndroidMapRenderer.legacy:
+        return 'legacy';
+      case AndroidMapRenderer.platformDefault:
+      case null:
+        break;
+    }
+    return 'unknown';
+  }
+
+  void _setMapId() {
+    setState(() {
+      _mapId = _mapIdController.text;
+
+      // Change key to initialize new map instance for new mapId.
+      _key = Key(_mapId ?? 'mapId#');
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final GoogleMap googleMap = GoogleMap(
+        onMapCreated: _onMapCreated,
+        initialCameraPosition: const CameraPosition(
+          target: _kMapCenter,
+          zoom: 7.0,
+        ),
+        key: _key,
+        cloudMapId: _mapId);
+
+    final List<Widget> columnChildren = <Widget>[
+      Padding(
+        padding: const EdgeInsets.all(10.0),
+        child: Center(
+          child: SizedBox(
+            width: 300.0,
+            height: 200.0,
+            child: googleMap,
+          ),
+        ),
+      ),
+      Padding(
+          padding: const EdgeInsets.all(10.0),
+          child: TextField(
+            controller: _mapIdController,
+            decoration: const InputDecoration(
+              hintText: 'Map Id',
+            ),
+          )),
+      Padding(
+          padding: const EdgeInsets.all(10.0),
+          child: ElevatedButton(
+            onPressed: () => _setMapId(),
+            child: const Text(
+              'Press to use specified map Id',
+            ),
+          )),
+      if (Platform.isAndroid)
+        Padding(
+          padding: const EdgeInsets.all(10.0),
+          child: Text(
+              'On Android, Cloud-based maps styling only works with "latest" renderer.\n\n'
+              'Current initialized renderer is "${_getInitializedsRendererType()}".'),
+        ),
+      if (Platform.isIOS)
+        const Padding(
+          padding: EdgeInsets.all(10.0),
+          child:
+              Text('On iOS, cloud based map styling works only if iOS platform '
+                  'version 12 or above is targeted in project Podfile. '
+                  "Run command 'pod update GoogleMaps' to update plugin"),
```

Comment:
These info texts are now removed, and example projects is forced to use iOS platform version 12:
https://github.com/flutter/packages/pull/3682/commits/a86315500e865263bad706002c9fafe94549ab8e
One problem with this could be that when flutter eventually starts using newer version (>12) of the iOS platform version by default, these example projects are still build with 12. 

Missing implementation for tests with old platform versions:
> Change the iOS implementation package to have multiple example projects so that we are testing compilation with all the SDK versions. I can look into doing this as a prequel PR; it's something I was thinking about recently in the context of another PR as well.

Reply:
My understanding is that if you don't explicitly set a platform version in your Podfile, CocoaPods will use the deployment target of your project bto build the pods, otherwise it will use the platform version set in the Podfile. Minimum deployment target should affect the cocoapod versions if the pods podspec file has s.ios.deployment_target set to certain version. I bet this has been done for GoogleMaps pod.
Just tested the build with ios11, ios12 and ios13 example apps, and feature seems to work on each as it should.